### PR TITLE
Fix column escaping with on ORDER BY

### PR DESCRIPTION
**Pull request by fpacifici**
When trying to ORDER BY a query by a column that contains a `.` this is what happens if we ORDER BY DESC:

```
ORDER BY (-events.last_seen AS -`events.last_seen`) DESC
```

This syntax is wrong. 
This happens for an interaction in column_expr between the logic that assigns the an alias to the column and the logic that escapes the alias:

-events.last_seen is interpreted as:
A column name to escape when generating the alias -> `events.last_seen`
But the escaping logic already takes the `-` out (the reason is not clear, it seems that should not happen at that point) -> -`events.last_seen`
since -events.last_seen and the escaped alias -`events.last_seen` are not the same string column_expr decides to create an alias thus the broken string above.

This is the smallest fix for the common case: it strips the '-' from the column nam,e and from the alias before applying escaping and aliasing and add it back later.
A good fix would be a complete redesign of the approach that would not pass the '-' character to column_expr at all, but that will be achieved through the new columns data model.

Test plan:
- added two unit tests for the corner cases
- the api tests still pass (with the feature on) both from snuba and from sentry
- I can run this query properly:
```
{
    "selected_columns": ["events.timestamp"],
    "orderby": ["-events.event_id"],
    "project": [1],
    "from_date": "2019-11-03T06:14:46",
    "to_date": "2019-11-08T06:14:46",
    "granularity": 3600
}
```

SNS-290

### CODE REVIEW

File: snuba/query/columns.py [hunk 1]
```diff
@@ -43,12 +45,40 @@ def column_expr(dataset, column_name, query: Query, parsing_context: ParsingCont
         return escape_literal(column_name[1:-1])
     else:
         expr = dataset.column_expr(column_name, query, parsing_context)
-
     if aggregate:
         expr = function_expr(aggregate, expr)
 
-    alias = escape_alias(alias or column_name)
-    return alias_expr(expr, alias, parsing_context)
+    strip_negation = state.get_config('process_alias_without_neg', 0)
```
Comment (fpacifici) [inline snuba/query/columns.py]: This is a precution during rollout since here there can be a vast number of corner cases. If everything works well I will remove the flag.

**Review by tkaemming — APPROVED** [review]
